pkg/server/handler/v1alpha1/workflowrun.go

```diff
@@ -59,6 +65,45 @@ func CreateWorkflowRun(ctx context.Context, project, workflow, tenant string, wf
 	return handler.K8sClient.CycloneV1alpha1().WorkflowRuns(common.TenantNamespace(tenant)).Create(wfr)
 }
 
+func workflowRunCreationDryRun(project, workflow, tenant string, wfr *v1alpha1.WorkflowRun) (*v1alpha1.WorkflowRun, error) {
```

Comment:
This function is a little confused, which will make `CreateWorkflowRun()` more complex.

The target of this function is to check the disk free space before create the workflowrun.
I think it would be better to add an additional API for workflowrun precheck. We can add more precheck logic into this API later if necessary.

Reply:
dry-run could do the same thing with precheck API. I think there is no need to add another API as the dry-run is easier to understand.